### PR TITLE
refactor: Expense Claim Cost Center

### PR DESCRIPTION
**Pull request by jiloysss**
Changed Script in Expense Claim (python) Where the cost center is based on the cost center field on each row in creating GL Entry

Filtered Cost Center in Expense Claim Detail where it will only fetch child accounts

Hid Cost Center in Expense Claim

Please read the pull request checklist to make sure your changes are merged: https://github.com/frappe/erpnext/wiki/Pull-Request-Checklist

### CODE REVIEW

**Comment by jiloysss** [comment]
EXPENSE CLAIM
* Added field Cost Center in table

expense_claim

EXPENSE CLAIM TYPE
* Added check box Deferred Expense Accounts
      - If checked, Default Account in table will be filtered with Asset Accounts. Expense Accounts if otherwise

Expense_claim_type

File: erpnext/hr/doctype/expense_claim/expense_claim.py [hunk 1]
```diff
@@ -127,7 +127,7 @@ def get_gl_entries(self):
 					"debit": data.sanctioned_amount,
 					"debit_in_account_currency": data.sanctioned_amount,
 					"against": self.employee,
-					"cost_center": self.cost_center
+					"cost_center": data.cost_center
```
Comment (deepeshgarg007) [inline erpnext/hr/doctype/expense_claim/expense_claim.py:130]: There is no field `cost_center` in expenses table thats why cost_center is not updated in GL Entry

File: erpnext/accounts/general_ledger.py [hunk 2]
```diff
@@ -124,6 +124,8 @@ def make_entry(args, adv_adj, update_outstanding, from_repost=False):
 	gle.insert()
 	gle.run_method("on_update_with_args", adv_adj, update_outstanding, from_repost)
 	gle.submit()
+	frappe.db.sql("""UPDATE `tabGL Entry` SET cost_center = %s WHERE name = %s """, (args.cost_center, gle.name))
```
Comment (deepeshgarg007) [inline erpnext/accounts/general_ledger.py:127]: This is very wrong. Pass cost center in GL entries itself. No need of explicit commit

**Review by deepeshgarg007 — CHANGES_REQUESTED** [review]

**Comment by deepeshgarg007** [comment]
@jiloysss Please rebase and raise PR for `version-11-hotfix` branch

**Pull request closed without merging**